File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder ans = new StringBuilder();
        int n = part.length();

        for (char c : s.toCharArray()) {
            ans.append(c);
            if (ans.length() >= n && ans.substring(ans.length() - n).equals(part)) {
                ans.setLength(ans.length() - n);
            }
        }
        return ans.toString();
    }
}






// class Solution {
//     public String removeOccurrences(String s, String part) {
//         Stack<Character> st = new Stack<>();
//         int n = part.length();

//         for (char ch : s.toCharArray()) {
//             st.push(ch);
//             if (st.size() >= n) {
//                 boolean match = true;
//                 for (int i = 0; i < n; i++) {
//                     if (st.get(st.size() - n + i) != part.charAt(i)) {
//                         match = false;
//                         break;
//                     }
//                 }
//                 if (match) {
//                     for (int i = 0; i < n; i++) {
//                         st.pop();
//                     }
//                 }
//             }
//         }

//         StringBuilder ans = new StringBuilder();
//         for (char ch : st) {
//             ans.append(ch);
//         }
//         return ans.toString();
//     }
// }






// // class Solution {
// //     public String removeOccurrences(String s, String part) {
// //         while (s.contains(part)) {
// //             s = s.replaceFirst(part, "");
// //         }
// //         return s;
// //     }
// // }

File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        st = []
        n = len(part)

        for ch in s:
            st.append(ch)
            if len(st) >= n and "".join(st[-n:]) == part:
                for _ in range(n):
                    st.pop()

        return "".join(st)




# class Solution:
#     def removeOccurrences(self, s: str, part: str) -> str:
#         while part in s:
#             s = s.replace(part, "", 1)
#         return s